server/api.js

```diff
@@ -4,15 +4,140 @@ import pool from "./db";
 
 const router = Router();
 
+// Get all the tasks
 router.get("/tasks", (req, res) => {
 	pool.query("SELECT * FROM tasks").then((result) => {
-		// console.log(result.rows);
 		return res.send(result.rows);
 	});
 });
-
 router.get("/", (_, res) => {
-	return res.json({ message: "Hello, from the api route!" });
+	res.json({ message: "Hello, from the api route!" });
 });
 
+router.post("/tasks", (req, res) => {
+	const { task_title, due_date, evidence, status_title, element_title } =
+		req.body;
+	let params = [];
+	if (!task_title) {
+		return res
+			.status(404)
+			.send({ success: false, message: "Task title cannot be empty" });
+	} else {
+		params.push(task_title);
+	}
+	if (due_date.length > 0) {
+		params.push(due_date);
+	} else {
+		params.push("");
+	}
+	if (evidence.length > 0) {
+		params.push(evidence);
+	} else {
+		params.push("");
+	}
+	if (status_title.length > 0) {
+		pool
+			.query("SELECT Status_id FROM status WHERE Status_title = $1", [
+				status_title,
+			])
+			.then((result) => {
+				params.push(result.rows[0].status_id);
+			})
+			.catch((err) => console.log(err));
+	} else {
+		res
+			.status(404)
+			.send({ success: false, message: "status_title cannot be empty" });
+	}
+	if (element_title.length > 0) {
+		pool
+			.query("SELECT Element_id FROM elements WHERE Element_title = $1", [
+				element_title,
+			])
+			.then((result) => {
+				params.push(result.rows[0].element_id);
+			})
+			.catch((err) => console.log(err));
+	} else {
+		res
+			.status(404)
+			.send({ success: false, message: "element_title cannot be empty" });
+	}
+
+	pool
+		.query(
+			"INSERT INTO tasks (Task_title,Due_date, Evidence, Status_id, Element_id) VALUES ($1,$2,$3,$4,$5)",
+			params
+		)
+		.then(() => {
+			pool.query("SELECT * FORM tasks").then((result) => res.send(result.rows));
+		});
+});
 export default router;
+
+//Delete a task
+
+router.delete("/:task_title", (req, res) => {
+	const title = req.params.task_title;
+	pool
+		.query("DELETE  FROM tasks WHERE Task_title = $1", [title])
+		.then(() => res.send(`Task ${title} deleted !`))
```

Comment:
Also what if the task_id doesn't exist in the database or if something wrong and unexpected happened, please account for that, if result.rows === 0 then we should send something back maybe?

Reply:
Can a primary key be null? PRIMARY KEY constraint combines the properties of both unique and not null

Reply:
No, you can't have a null primary key. But if the user tries to delete a task that doesn't exist, they should get a 404 Not Found.